lib/evmone/baseline.cpp

```diff
@@ -109,25 +112,25 @@ inline evmc_status_code check_requirements(
 }
 
 template <bool TracingEnabled>
-evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& analysis) noexcept
+evmc_result execute(const VM& vm, ExecutionState& state, const EOF1Header& header,
+    const CodeAnalysis& analysis) noexcept
 {
-    // Use padded code.
-    state.code = {analysis.padded_code.get(), state.code.size()};
-
     auto* tracer = vm.get_tracer();
     if constexpr (TracingEnabled)
         tracer->notify_execution_start(state.rev, *state.msg, state.code);
 
     const auto& instruction_table = get_baseline_instruction_table(state.rev);
 
-    const auto* const code = state.code.data();
-    auto pc = code;
-    while (true)  // Guaranteed to terminate because padded code ends with STOP.
+    // Use padded code for pc.
+    // CODECOPY/CODESIZE will still use original code from state.code
+    const auto* const padded_code = analysis.padded_code.get();
+    const auto* pc = padded_code + header.code_begin();
```

Comment:
Somewhat hacky we're setting here PC to padded code without changing `state.code`. But `CODECOPY` needs to see original code with data section.